### PR TITLE
Featured article fix

### PR DESCRIPTION
**Pull request by milosyuki**
Added ability for a featured post to be switched off. In that case, the first post becomes featured.
Screenshot 2023-05-10 at 23 10 44

### CODE REVIEW

File: _data/settings/blog-posts.yml [hunk 1]
```diff
@@ -1 +1,2 @@
+show_featured_post: true
 featured_post: e449aae9-628a-47a4-b5eb-d02644f95c25
```
Comment (thecotne) [inline _data/settings/blog-posts.yml]: can't we just make "featured_post" nullable/optional field? @milosyuki

File: workspaces/cms-scripts/src/data.ts [hunk 2]
```diff
@@ -106,6 +107,7 @@ export async function fileToPost(
     sourceFilepath,
     gitlog: await gitlog(sourceFilepath),
     featured: blogPosts.featured_post === data.id,
+    show_featured_post: data.show_featured_post,
```
Comment (thecotne) [inline workspaces/cms-scripts/src/data.ts]: this should be 

```
    show_featured_post: blogPosts.show_featured_post,
```

Comment (thecotne) [reply workspaces/cms-scripts/src/data.ts]: but also why would we have that boolean on every post?

File: workspaces/cms-data/tsconfig.json [hunk 3]
```diff
@@ -4,6 +4,6 @@
     "outDir": ".types",
     "lib": ["DOM", "DOM.Iterable", "ESNext"]
   },
-  "include": ["src/**/*"],
+  "include": ["src/**/*", "src/settings/blog-posts.ts"],
```
Comment (thecotne) [inline workspaces/cms-data/tsconfig.json]: don't add this

File: tsconfig.json [hunk 4]
```diff
@@ -28,7 +28,7 @@
     "./vite-env.d.ts",
     "./src/**/*",
     ".next/types/**/*.ts"
-  ],
+, "workspaces/cms-data/src/settings/blog-posts.ts"  ],
```
Comment (thecotne) [inline tsconfig.json]: pretty sure this is not needed

File: src/app/[locale]/posts/(components)/PostsPage.tsx [hunk 5]
```diff
@@ -317,26 +316,39 @@ interface Block {
...
+function CustomHits({ categories, params }: Pick<Props, "categories" | "params">) {
+  const [blogPostsSettings, setBlogPostsSettings] = useState<any>();
+  useEffect(() => {
+    async function fetchData() {
+      const response = await fetch(`/api/blogPosts?locale=${params.locale}`);
+      const data = await response.json();
+      setBlogPostsSettings(data);
+    }
+    fetchData();
+  }, []);
```
Comment (thecotne) [inline src/app/[locale]/posts/(components)/PostsPage.tsx]: we have not used this kind of fetching logic on any other data why is this different?

**Review by thecotne — APPROVED** [review]

**Review by robwalsh21 — APPROVED** [review]

**Pull request merged**